### ISSUE TITLE
Can't use queries as statement names

### ISSUE THREAD

**Issue by essen** [ostinelli/pgpool#3]
It causes conflicts because Postgres will truncate them. This is what a PREPARE query with a statement name that is too long says:

```
NOTICE:  identifier "SELECT * FROM orgs NATURAL INNER JOIN projects NATURAL INNER JOIN bugs NATURAL INNER JOIN comments WHERE org_name=$1 AND project_name=$2 AND project_bug_id=$3 AND comment_id=$4" will be truncated to "SELECT * FROM orgs NATURAL INNER JOIN projects NATURAL INNER JO"
```

I ran into this because I have two such queries that have the same start and pgpool/Postgres disagree on it existing already. This leads to:

```
** {{badmatch,{error,{error,error,<<"42P05">>,duplicate_prepared_statement,
```

Maybe use something like `<<"statement-", (integer_to_binary(erlang:phash2(<<"c">>)))/binary>>` for the statement name on the Postgres side. Will experiment and send a PR.

**Comment by ostinelli** [comment]
Hi Loïc,
This is a valid error though. I wouldn't want to obfuscate the prepared statement name at library level. What you are doing makes a lot of sense from an application perspective, but IMO not a library level.

Maybe I've missed a concern?

**Comment by essen** [comment]
???

I have two different SQL queries that just happen to start the same way, and I can't use them both with pgpool because Postgres truncates the statement name and thinks the second query already exists (while pgpool doesn't truncate).

A simplified example:

| Query | pgpool statement name | Postgres statement name |
| - | - | - |
| SELECT 1 | SELECT 1 | SELEC |
| SELECT 2 | SELECT 2 | SELEC |

Only in the real world Postgres seems to truncate at around 32 bytes, not 5 like in this simplified example.

**Comment by ostinelli** [comment]
I understood what you meant, but I forgot that PgPool does the prepared statement optimization under the hood (I though you were naming them yourself).

What you are suggesting can probably be applied [here](https://github.com/ostinelli/pgpool/blob/1.1.1/src/pgpool_worker.erl#L290). Are you working on a PR? Otherwise I'll do so in the next days.

**Comment by essen** [comment]
Right.

I have a working patch, I'll send it when I get back to this project.

**Comment by ostinelli** [comment]
Fixed in `1.2.0`.
